apps/server/src/modules/task/service/submission.service.ts

```diff
@@ -20,10 +30,38 @@ export class SubmissionService {
 		return submissions;
 	}
 
+	async findByUserAndTask(userId: EntityId, taskId: EntityId): Promise<Submission[]> {
+		const [submissions] = await this.submissionRepo.findAllByUserId(userId);
+		const filteredSubmissions = submissions.filter((submission) => submission.task.id === taskId);
+
+		return filteredSubmissions;
+	}
+
 	async delete(submission: Submission): Promise<void> {
 		const params = FileParamBuilder.build(submission.school.id, submission);
 		await this.filesStorageClientAdapterService.deleteFilesOfParent(params);
 
 		await this.submissionRepo.delete(submission);
 	}
+
+	async createForTaskCard(userId: EntityId, taskId: EntityId): Promise<Submission> {
```

Comment:
I would rename into createSubmissionForUser or create empty submission or something like that 
because we create submission for the user, not for the task card.
There is nothing in code indicating task-card here, other than the fact that is called from task-card uc.
But the name should be more generic